src/borg/archiver.py

```diff
@@ -1145,12 +1145,16 @@ def _delete_archives(self, args, repository):
         stats = Statistics()
         with Cache(repository, key, manifest, progress=args.progress, lock_wait=self.lock_wait) as cache:
             for i, archive_name in enumerate(archive_names, 1):
-                msg = 'Would delete archive: {} ({}/{})' if dry_run else 'Deleting archive: {} ({}/{})'
-                logger.info(msg.format(format_archive(manifest.archives[archive_name]), i, len(archive_names)))
-                if not dry_run:
-                    archive = Archive(repository, key, manifest, archive_name, cache=cache,
-                                      consider_part_files=args.consider_part_files)
-                    archive.delete(stats, progress=args.progress, forced=args.forced)
+                archive = Archive(repository, key, manifest, archive_name, cache=cache,
+                                  consider_part_files=args.consider_part_files)
+                try:
+                    msg = 'Would delete archive: {} ({}/{})' if dry_run else 'Deleting archive: {} ({}/{})'
+                    logger.info(msg.format(format_archive(manifest.archives[archive_name]), i, len(archive_names)))
+                    if not dry_run:
+                        archive.delete(stats, progress=args.progress, forced=args.forced)
+                except NoArchiveError:
+                    logger.warning('Archive {} not found ({}/{}).'.format(archive_name, i, len(archive_names)))
+
```

Comment:
well, this is now a mix up of my idea how to solve the "too wide scope for just catching KeyError" and your idea.

mine was to rely on Archive class bailing out when the archive name is not known.

yours was to make a more specific subclass of KeyError, which would somehow solve the problem of wide-scope catching a KeyError that could be caused by a lot of code (while not killing the loop).

but in the end, we do not even need the more specific subclass, but we need narrow-scope KeyError catching:

```
msg_delete = ...
msg_not_found = ...
for ...
    try:
        archive_info = manifest.archives[archive_name]
    except KeyError:  # note the very close scope, KeyError can only happen for what we really mean
        logger.warning(msg_not_found...)
    else:
        logger.info(msg_delete...)
        ...
```